pymodbus/transport/transport.py

```diff
@@ -490,8 +489,10 @@ async def do_reconnect(self) -> None:
                     self.comm_params.reconnect_delay_max,
                 )
         except asyncio.CancelledError:
-            pass
-        self.reconnect_task = None
+            self.reconnect_delay_current = self.comm_params.reconnect_delay or 0.0
```

Comment:
This is wrong.

Reply:
> This is wrong.

Should this be in the finally block instead?

> I am also surprised that you do not re-raise the exception....you continue suppressing.

But I do re-raise the expression, right below this line...